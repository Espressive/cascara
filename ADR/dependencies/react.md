# React

Our React version is tied to FE's version, we can bump only after they are doing bumping Ract. For now, all the following ecosystem members will be pinned to their current version:

- @types/react v16.8.0
- @types/react-dom v16.8.0
- react v16.13.1
- react-dom v16.13.1
- react-router v5.2.0
- react-router-dom v5.2.0
- react-scripts v4.0.3
- react-test-renderer v16.13.1
