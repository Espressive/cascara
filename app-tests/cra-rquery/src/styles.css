html,
body,
#login {
  margin: 0.5em;
  padding: 0;
}

main {
  display: grid;
  justify-content: center;
  align-content: center;
  background-color: ghostwhite;
}

aside {
  grid-column: 1;
}

aside nav {
  display: flex;
  flex-direction: column;
}

aside nav a {
  line-height: 1.5em;
}

section {
  grid-column: 2;
}
