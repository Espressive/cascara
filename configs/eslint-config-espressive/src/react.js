// "off" or 0 - turn the rule off
// "warn" or 1 - turn the rule on as a warning (doesn't affect exit code)
// "error" or 2 - turn the rule on as an error (exit code will be 1)

module.exports = {
  rules: {
    'react/boolean-prop-naming': 2,
    'react/default-props-match-prop-types': 2,
    'react/destructuring-assignment': 2,
    'react/jsx-fragments': 2,
    'react/jsx-no-duplicate-props': 2,
    'react/jsx-no-useless-fragment': 2,
    'react/jsx-pascal-case': 2,
    'react/jsx-sort-default-props': 2,
    'react/jsx-sort-props': 2,
    'react/no-children-prop': 2,
    'react/no-danger': 2,
    'react/no-find-dom-node': 2,
    'react/no-multi-comp': 2,
    'react/no-string-refs': 2,
    'react/no-this-in-sfc': 2,
    'react/no-typos': 2,
    'react/no-unused-prop-types': 2,
    'react/no-unused-state': 2,
    'react/prefer-stateless-function': 2,
    'react/self-closing-comp': 2,
    'react/sort-prop-types': 2,
    'react/style-prop-object': 2,
  },
};
