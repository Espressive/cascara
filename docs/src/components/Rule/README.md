# Rule

This component is used to define rules for our design system. Images to support the rule are optional.
