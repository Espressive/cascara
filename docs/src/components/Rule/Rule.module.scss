// Shared styles for the rule images
%ruleImg {
  h6 {
    font-weight: 600;
    font-size: 1rem;

    &:before {
      margin-right: 0.5em;
    }
  }
}

._ {
  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    margin-top: 0;
    margin-bottom: 0.25em;
  }

  .Do {
    @extend %ruleImg;
    color: green;

    h6:before {
      content: '✔️';
    }
  }

  .Dont {
    @extend %ruleImg;
    color: red;

    h6:before {
      content: '🚫';
    }
  }
}
