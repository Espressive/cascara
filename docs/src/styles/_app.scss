html[data-layout='admin'] div#__next {
  --brand-primary-color: #b4104b;
}

a {
  color: var(--brand-primary-color);
  &:hover {
    color: var(--brand-primary-color);
    text-decoration: underline;
  }
}

:not(pre) > code {
  padding: 0.125em 0.25em;
  border-radius: 0.25em;
  color: rgb(255, 245, 246);
  text-shadow: rgb(0, 0, 0) 0px 0px 2px, rgba(252, 31, 44, 0.46) 0px 0px 10px,
    rgba(252, 31, 44, 0.46) 0px 0px 5px, rgba(252, 31, 44, 0.46) 0px 0px 25px;
  background-color: #111;
  border: 1px solid rgba(0, 0, 0, 0.25);
}

dl {
  display: grid;
  grid-template-columns: max-content auto;
  grid-row-gap: 1ch;

  dt {
    font-weight: 600;
    font-size: 0.9em;
    grid-column-start: 1;
  }

  dd {
    grid-column-start: 2;

    * + * {
      margin-left: 0.5em;
    }

    span {
      &:after {
        content: ',';
      }

      &:last-child:after {
        content: ' ';
      }
    }
  }
}

main {
  & > div {
    // max-width: 60em;
    padding: 2em;
  }

  img.right {
    float: right;
    margin-left: 1em;
  }

  img.left {
    float: left;
    margin-right: 1em;
  }

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    a {
      margin-left: -1.125em;
      display: inline-block;
      width: 1.125em;
      height: 1em;
      text-align: center;
      opacity: 0;
      transform: translateX(100%);
      transition: all 200ms ease-in-out;
      outline: none;
      background-image: url('../img/link.svg');
      background-repeat: no-repeat;
      background-position: left center;
      background-size: 90% 90%;
    }

    &:hover a {
      opacity: 1;
      transform: translateX(0);

      &:hover {
        color: var(--document-text);
        text-decoration: none;
      }
    }
  }

  p {
    line-height: 1.75em;
  }

  ul,
  ol {
    line-height: 2em;
    padding-inline-start: 2em;
  }

  blockquote {
    margin-inline-start: 0;
    margin-inline-end: 0;
    padding: 0.5em 1em;
    background-color: var(--color-transparent-grey, grey);
    border-radius: 0.25em;
    border-inline-start: 0.25em solid var(--brand-primary-color);

    footer {
      &:before {
        content: '- ';
      }
      text-align: right;
      font-style: italic;
    }
  }
}
