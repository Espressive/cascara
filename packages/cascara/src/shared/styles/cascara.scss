@forward 'lib/scroll' as scroll-*;
@forward 'color/brand';

$test-var: var(--test);
