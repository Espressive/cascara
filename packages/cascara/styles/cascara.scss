@forward '@espressive/design-tokens/scss';
@forward 'breakpoints' as media-*;
@forward 'scroll' as scroll-*;
