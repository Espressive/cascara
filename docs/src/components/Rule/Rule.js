import React from 'react';
import pt from 'prop-types';
import { Columns, Image, Section } from '@espressive/cascara';
import styles from './Rule.module.scss';

const propTypes = {
  body: pt.string.isRequired,
  doImg: pt.string,
  dontImg: pt.string,
  title: pt.string.isRequired,
};

const Rule = ({ title, body, doImg, dontImg }) => {
  return (
    <Section className={styles._} title={title}>
      <p>{body}</p>
      <Columns count={2}>
        {dontImg && (
          <div className={styles.Dont}>
            <h6>Don&apos;t Do This</h6>
            <Image
              alt={`Incorrect usage example for: ${title}`}
              src={dontImg}
            />
          </div>
        )}
        {doImg && (
          <div className={styles.Do}>
            <h6>Do This</h6>
            <Image alt={`Correct usage example for: ${title}`} src={doImg} />
          </div>
        )}
      </Columns>
    </Section>
  );
};

Rule.propTypes = propTypes;

export default Rule;
