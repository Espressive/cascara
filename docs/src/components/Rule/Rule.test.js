import React from 'react';
import { render, screen } from '@testing-library/react';

import Rule from './Rule';

describe('<Rule />', () => {
  // run snapshot test for all fixtures.
  test('empty snapshot', async () => {
    const view = render(
      <Rule body='Required Body' title='Required Title' />
    ).container;
    expect(view).toMatchSnapshot();
  });

  test('renders accessible images', async () => {
    const ruleTitle = 'My Rule';
    const ruleImageSrc = 'https://placedog.net/500/500';

    const view = render(
      <Rule
        body='Required body'
        doImg={ruleImageSrc}
        dontImg={ruleImageSrc}
        title={ruleTitle}
      />
    );
    expect(view).toMatchSnapshot();
    expect(
      screen.getByAltText(`Incorrect usage example for: ${ruleTitle}`)
    ).toBeVisible();
    expect(
      screen.getByAltText(`Correct usage example for: ${ruleTitle}`)
    ).toBeVisible();
  });
});
