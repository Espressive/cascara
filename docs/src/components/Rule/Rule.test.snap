// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rule /> empty snapshot 1`] = `
<div>
  <section
    class="☕️_Section____qiuc5 ☕️_Section____qiuc5"
    data-component="Section"
  >
    <h3
      class="☕️_Title_Title__18epm"
    >
      Required Title
       
    </h3>
    <p>
      Required Body
    </p>
    <div
      class="☕️_Columns____w6xf6 ☕️_Columns_fixed__w6xf6"
      data-component="Columns"
      style="column-count: 2;"
    />
  </section>
</div>
`;

exports[`<Rule /> renders accessible images 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="☕️_Section____qiuc5 ☕️_Section____qiuc5"
        data-component="Section"
      >
        <h3
          class="☕️_Title_Title__18epm"
        >
          My Rule
           
        </h3>
        <p>
          Required body
        </p>
        <div
          class="☕️_Columns____w6xf6 ☕️_Columns_fixed__w6xf6"
          data-component="Columns"
          style="column-count: 2;"
        >
          <div
            class="Dont"
          >
            <h6>
              Don't Do This
            </h6>
            <img
              alt="Incorrect usage example for: My Rule"
              class="☕️_Image____1q5in "
              src="https://placedog.net/500/500"
            />
          </div>
          <div
            class="Do"
          >
            <h6>
              Do This
            </h6>
            <img
              alt="Correct usage example for: My Rule"
              class="☕️_Image____1q5in "
              src="https://placedog.net/500/500"
            />
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="☕️_Section____qiuc5 ☕️_Section____qiuc5"
      data-component="Section"
    >
      <h3
        class="☕️_Title_Title__18epm"
      >
        My Rule
         
      </h3>
      <p>
        Required body
      </p>
      <div
        class="☕️_Columns____w6xf6 ☕️_Columns_fixed__w6xf6"
        data-component="Columns"
        style="column-count: 2;"
      >
        <div
          class="Dont"
        >
          <h6>
            Don't Do This
          </h6>
          <img
            alt="Incorrect usage example for: My Rule"
            class="☕️_Image____1q5in "
            src="https://placedog.net/500/500"
          />
        </div>
        <div
          class="Do"
        >
          <h6>
            Do This
          </h6>
          <img
            alt="Correct usage example for: My Rule"
            class="☕️_Image____1q5in "
            src="https://placedog.net/500/500"
          />
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
